core/src/main/scala/hedgehog/Gen.scala

```diff
@@ -121,20 +121,25 @@ trait GenTOps extends MonadGenOps[Gen] {
     */
    def frequency[A](a: (Int, GenT[A]), l: List[(Int, GenT[A])]): GenT[A] = {
      @annotation.tailrec
-     def pick(n: Int, x: (Int, GenT[A]), xs: List[(Int, GenT[A])]): GenT[A] =
+     def pick(n: Long, x: (Int, GenT[A]), xs: List[(Int, GenT[A])]): GenT[A] =
        if (n <= x._1)
          x._2
        else
          xs match {
            case Nil =>
-             sys.error("Invariant: frequency hits an impossible code path")
+             Gen.discard
```

Comment:
Firstly, thank you so much for all the code contributions!!! I really appreciate it, and the time you've taken to do the work.

Just one small question with this otherwise awesome PR. As much as I hate the previous exception throwing, I do think it can help with debugging, at least until we have something better. `discard` (at the moment) is really intended for infrequent discards around conditions. At least in how I have seen/used `frequency` the weights are static, and so it's an invariant to hit this line, and having the line number and stack trace can really help, instead of a very vague discard error.

Did you mind leaving this in for now?

You mind be interested in this issue, which I think we should do in the scala version too, and which might make me less reluctant to use dicard.

https://github.com/hedgehogqa/haskell-hedgehog/issues/335

Reply:
> Ok I have put it back now.

Thanks!

> What do you think about when non-positive weights are given in the first place? 

Yeah sorry I didn't say anything about that sooner. Ideally they're an invariant, and an error picks it up staight away. If you didn't mind changing that while you're there that would be great!